include/wil/registry_basic.h

```diff
@@ -1531,133 +1528,123 @@ inline HRESULT get_value_dword_nothrow(HKEY key, _In_opt_ PCWSTR value_name, _In
 inline HRESULT get_value_qword_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, _In_ DWORD64* return_value) WI_NOEXCEPT
 {
     DWORD data_size_bytes{sizeof *return_value};
-    return HRESULT_FROM_WIN32(RegGetValueW(key, subkey, value_name, ::wil::registry::details::get_value_flags_from_value_type(REG_QWORD), nullptr, return_value, &data_size_bytes));
+    return HRESULT_FROM_WIN32(RegGetValueW(key, subkey, value_name, ::wil::reg::details::get_value_flags_from_value_type(REG_QWORD), nullptr, return_value, &data_size_bytes));
 }
 
 inline HRESULT get_value_qword_nothrow(HKEY key, _In_opt_ PCWSTR value_name, _In_ DWORD64* return_value) WI_NOEXCEPT
 {
     return get_value_qword_nothrow(key, nullptr, value_name, return_value);
 }
 
-// TODO: write value_byte_vector getters that don't throw.
-//inline HRESULT get_value_byte_vector_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, DWORD type, ::std::vector<BYTE>& data) WI_NOEXCEPT
-//{
-//    const reg_view_details::reg_view_nothrow regview{key};
-//    optional_value<::std::vector<BYTE>> optionalvalue;
-//    const auto hr = regview.try_get_value<::std::vector<BYTE>>(subkey, value_name, optionalvalue, type);
-//    RETURN_IF_FAILED(hr);
-//
-//    if (optionalvalue.type != optional_value_status::has_value)
-//    {
-//        RETURN_WIN32(ERROR_FILE_NOT_FOUND);
-//    }
-//
-//    std::swap(data, optionalvalue.value);
-//    return S_OK;
-//}
-//
-//inline HRESULT get_value_byte_vector_nothrow(HKEY key, _In_opt_ PCWSTR value_name, DWORD type, ::std::vector<BYTE>& data) WI_NOEXCEPT
-//{
-//    return get_value_byte_vector_nothrow(key, nullptr, value_name, type, data);
-//}
+#if defined(__WIL_WINREG_STL)
+inline HRESULT get_value_byte_vector_nothrow(HKEY key, _In_opt_ PCWSTR subkey, _In_opt_ PCWSTR value_name, DWORD type, ::std::vector<BYTE>* data) WI_NOEXCEPT
+try
+{
+    const reg_view_details::reg_view_nothrow regview{ key };
+    ::wil::reg::optional_value<::std::vector<BYTE>> registry_value {regview.try_get_value<::std::vector<BYTE>>(subkey, value_name, type)};
+    RETURN_IF_FAILED(registry_value.inner_error);

```

Comment:
I'm not sure I like the `inner_error` concept---I'd much rather use a straightforward `std::optional` as an outparam, then have the function return an HRESULT.

This seems a lot like Rust's `Result` type, which is really cool, but I think it's a bit non-standard for C++. I'd rather stick to common paradigms and use std-like types whenever possible.